### PR TITLE
Ensure --version works with non-ascii project path

### LINKED ISSUE #66617: Ansible does not work when the working directory contains some non-ASCII characters
##### SUMMARY
As of Ansible 2.9, the Ansible CLI commands crashe when the ansible project directory contains some non-ASCII characters. In my case it contains a `é`, a quite common character in french.

I get this error:

> ERROR! Unexpected Exception, this is probably a bug: 'ascii' codec can't decode byte 0xc3 in position 31: ordinal not in range(128)
> to see the full traceback, use -vvv

With `---v`, I get that:

> ERROR! Unexpected Exception, this is probably a bug: 'ascii' codec can't decode byte 0xc3 in position 31: ordinal not in range(128)
> the full traceback was:
> 
> Traceback (most recent call last):
>   File "/usr/bin/ansible-playbook", line 123, in <module>
>     exit_code = cli.run()
>   File "/usr/lib/python2.7/dist-packages/ansible/cli/playbook.py", line 69, in run
>     super(PlaybookCLI, self).run()
>   File "/usr/lib/python2.7/dist-packages/ansible/cli/__init__.py", line 82, in run
>     display.vv(to_text(opt_help.version(self.parser.prog)))
>   File "/usr/lib/python2.7/dist-packages/ansible/cli/arguments/option_helpers.py", line 174, in version
>     result += "\n  config file = %s" % C.CONFIG_FILE
> UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 31: ordinal not in range(128)

If I remove this character from the directory path, it gets working.

Note that this also happens for other Ansible commands, such as `ansible --version`. It sounds like it happens as soon as the project configuration is read.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

Ansible.

##### ANSIBLE VERSION

```
ansible 2.9.3
  config file = /tmp/project/ansible.cfg
  configured module search path = [u'/home/username/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/dist-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.17 (default, Nov  7 2019, 10:07:09) [GCC 7.4.0]
```

##### CONFIGURATION

```
ANSIBLE_PIPELINING(/tmp/project/ansible.cfg) = True
DEFAULT_HOST_LIST(/tmp/project/ansible.cfg) = [u'/tmp/project/hosts']
DEFAULT_REMOTE_USER(/tmp/project/ansible.cfg) = username
```

##### OS / ENVIRONMENT

Ubuntu 18.04. Ansible installed from the [Ansible PPA](https://launchpad.net/~ansible).

##### STEPS TO REPRODUCE
Create any ansible project in a directory containing a non-ASCII character.

##### EXPECTED RESULTS
Ansible should work and used to work until Ansible 2.9.

##### ACTUAL RESULTS
Get the error reported above.

```paste below

```

### PR DESCRIPTION
**Pull request by mkrizek**
##### SUMMARY
Fixes #66617

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

`lib/ansible/config/manager.py`

### CODE REVIEW

File: lib/ansible/cli/arguments/option_helpers.py [hunk 1]
```diff
@@ -164,22 +164,23 @@ def _gitinfo():
...
-    result = result + "\n  configured module search path = %s" % cpath
-    result = result + "\n  ansible python module location = %s" % ':'.join(ansible.__path__)
-    result = result + "\n  executable location = %s" % sys.argv[0]
-    result = result + "\n  python version = %s" % ''.join(sys.version.splitlines())
+        cpath = u":".join(C.DEFAULT_MODULE_PATH)
+    python_module_location = u":".join(map(to_text, ansible.__path__))
+    result = result + u"\n  configured module search path = %s" % cpath
+    result = result + u"\n  ansible python module location = %s" % python_module_location
+    result = result + u"\n  executable location = %s" % to_text(sys.argv[0])
+    result = result + u"\n  python version = %s" % ''.join(sys.version.splitlines())
```
Comment (sivel) [inline lib/ansible/cli/arguments/option_helpers.py]: Should we have a `u` on the join too?  Also, does `sys.version.splitlines()` produce text? Should we be explicit there if not?

Comment (mkrizek) [reply lib/ansible/cli/arguments/option_helpers.py]: Thanks, I completely missed the last `join`. Yes, it needs `u` and to be converted to text. Also, just noticed that `__version__` ~needs to be converted as well~ is `str` and might be better convert it too?

Comment (mkrizek) [reply lib/ansible/cli/arguments/option_helpers.py]: Probably does not make sense to convert `__version__`.

**Comment by samdoran** [comment]
Please add an integration test for this, then we can merge it.

**Comment by mkrizek** [comment]
Decided to turn this into a minimal fix for #66617.

**Review by samdoran — APPROVED** [review]

**Comment by samdoran** [comment]
rebuild_merge

**Pull request merged**